node/external/blockAPI/apiBlockFactory_test.go

```diff
@@ -0,0 +1,409 @@
+package blockAPI
+
+import (
+	"encoding/hex"
+	"encoding/json"
+	"math/big"
+	"testing"
+
+	"github.com/ElrondNetwork/elrond-go-core/core"
+	"github.com/ElrondNetwork/elrond-go-core/data/api"
+	"github.com/ElrondNetwork/elrond-go-core/data/block"
+	"github.com/ElrondNetwork/elrond-go/dataRetriever"
+	"github.com/ElrondNetwork/elrond-go/node/mock"
+	"github.com/ElrondNetwork/elrond-go/process/txstatus"
+	"github.com/ElrondNetwork/elrond-go/storage"
+	"github.com/ElrondNetwork/elrond-go/testscommon/dblookupext"
+	"github.com/stretchr/testify/assert"
+)
+
+func createMockArgsAPIBlockProc() *ArgAPIBlockProcessor {
+	statusComputer, _ := txstatus.NewStatusComputer(0, mock.NewNonceHashConverterMock(), &mock.ChainStorerStub{})
+	return &ArgAPIBlockProcessor{
+		Store:                    &mock.ChainStorerStub{},
+		Marshalizer:              &mock.MarshalizerFake{},
+		Uint64ByteSliceConverter: mock.NewNonceHashConverterMock(),
+		HistoryRepo:              &dblookupext.HistoryRepositoryStub{},
+		UnmarshalTxHandler:       &mock.TransactionAPIHandlerStub{},
+		StatusComputer:           statusComputer,
+		Hasher:                   &mock.HasherMock{},
+		AddressPubkeyConverter:   &mock.PubkeyConverterMock{},
+	}
+}
+
+func TestGetBlockByHash_NilUint64ByteSliceConverterShouldErr(t *testing.T) {
```

Comment:
i think the test's name doesn't resemble its actions

Reply:
renamed.